src/main/java/org/example/MediumAI.java

```diff
@@ -0,0 +1,53 @@
+package org.example;
+
+import static tictactoe.Main.*;
+
+public class MediumAI extends Player{
+    @Override
+    public char[][] play(char[][] matrix, char yourChar) {
+        if (checkWinner(matrix).equals("Game not finished")) {
+            System.out.println("Making move level \"medium\"");
+            //1st rule
```

Comment:
hmmm, honestly if code is written well and namings are all spot on, there is no need for comments. If you think these 3 are kinda different logical steps, extract them in 3 different private methods with good naming.